src/core/unit.c

```diff
@@ -3612,37 +3612,78 @@ static int get_name_owner_handler(sd_bus_message *message, void *userdata, sd_bu
 }
 
 int unit_install_bus_match(Unit *u, sd_bus *bus, const char *name) {
+        _cleanup_(sd_bus_message_unrefp) sd_bus_message *m1 = NULL, *m2 = NULL;
+        Service *s = SERVICE(u);
```

Comment:
please don't. we try to avoid doing something like this: cast in generic code to the specific type.

first of all, it's kinda ugly to assume that the unit this is called for is necessarily a service (it will be, because we are not using the bus name watch logic for any other unit types currently, but this might not stay that way).

But more importantly it's a layer violation: let's not stick unit-type-specific code in the generic sources. There's a reason why unit.c and service.c are two distinct code files.

hence, either pass thie timeout in as a function param. or if this doesn't work (because there are codepaths where the stackframe never includes service.c code that would acces it), then please add add a new vtable function to UnitVTable that returns the start timeout. that way generic code can query the unit type implementation which timeout to apply.